concent_api/core/payments/bankster.py

```diff
@@ -147,3 +157,130 @@ def finalize_payment(
         )
 
     return requestors_claim_payment_info, providers_claim_payment_info
+
+
+def settle_overdue_acceptances(
+    requestor_ethereum_address: str,
+    provider_ethereum_address: str,
+    acceptances: List[SubtaskResultsAccepted],
+) -> ClaimPaymentInfo:
+    """
+    The purpose of this operation is to calculate the total amount that the requestor owes provider for completed
+    computations and transfer that amount from requestor's deposit.
+    The caller is responsible for making sure that the payment is legitimate and should be performed.
+    Bankster simply calculates the amount and executes it.
+    """
+
+    assert isinstance(requestor_ethereum_address, str)
+    assert isinstance(provider_ethereum_address, str)
+    assert all([isinstance(acceptance, SubtaskResultsAccepted) for acceptance in acceptances])
+
+    assert len(requestor_ethereum_address) == ETHEREUM_ADDRESS_LENGTH
+    assert len(provider_ethereum_address) == ETHEREUM_ADDRESS_LENGTH
+    assert provider_ethereum_address != requestor_ethereum_address
+
+    # Bankster asks SCI about the amount of funds available in requestor's deposit.
+    requestor_deposit_value = service.get_deposit_value(client_eth_address=requestor_ethereum_address)
+
+    # Concent defines time T0 equal to oldest payment_ts from passed SubtaskResultAccepted messages from
+    # subtask_results_accepted_list.
+    oldest_payments_ts = min(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
+
+    current_time = get_current_utc_timestamp()
+
+    # Concent gets list of transactions from payment API where timestamp >= T0.
+    list_of_transactions = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+        requestor_eth_address=requestor_ethereum_address,
+        provider_eth_address=provider_ethereum_address,
+        payment_ts=oldest_payments_ts,
+        current_time=current_time,
+        transaction_type=TransactionType.BATCH,
+    )
+
+    # Concent defines time T1 equal to youngest timestamp from list of transactions.
+    if len(list_of_transactions) == 0:
+        t1_is_bigger_than_payments_ts = None
+    else:
+        youngest_transaction = max(transaction.closure_time for transaction in list_of_transactions)
+        # Concent checks if all passed SubtaskResultAccepted messages from subtask_results_accepted_list
+        # have payment_ts < T1
+        t1_is_bigger_than_payments_ts = any(
+            youngest_transaction > subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances
+        )
+
+    # Any of the items from list of overdue acceptances
+    # matches condition current_time < payment_ts + PAYMENT_DUE_TIME
+    acceptance_time_overdue = any(
+        current_time < subtask_results_accepted.payment_ts + settings.PAYMENT_DUE_TIME
+        for subtask_results_accepted in acceptances
+    )
+
+    if t1_is_bigger_than_payments_ts or acceptance_time_overdue:
```

Comment:
as we are using variables in logical context, both of them should be of type: `bool`. `t1_is_bigger_than_payments_ts` can be `None`

Reply:
:+1: done